tmail-backend/guice/distributed/src/test/java/com/linagora/tmail/event/BlobEventSerializationTest.java

```diff
@@ -0,0 +1,77 @@
+package com.linagora.tmail.event;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BucketName;
+import org.apache.james.blob.api.PlainBlobId;
+import org.apache.james.events.Event;
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.linagora.tmail.blob.secondaryblobstore.FailedBlobEvents;
+import com.linagora.tmail.blob.secondaryblobstore.ObjectStorageIdentity;
+
+public class BlobEventSerializationTest {
+    static final BlobId.Factory BLOB_ID_FACTORY = new PlainBlobId.Factory();
+    static final Event.EventId EVENT_ID = Event.EventId.of("6e0dd59d-660e-4d9b-b22f-0354479f47b4");
+    public static final BucketName BUCKET_NAME = BucketName.of("bucket-1");
+    public static final BlobId BLOB_ID = BLOB_ID_FACTORY.of("blob-id-1");
+    static final Event BLOB_ADDTITION_EVENT = new FailedBlobEvents.BlobAddition(EVENT_ID, BUCKET_NAME, BLOB_ID, ObjectStorageIdentity.PRIMARY);
+    static final Event BLOB_DELETION_EVENT = new FailedBlobEvents.BlobDeletion(EVENT_ID, BUCKET_NAME, BLOB_ID, ObjectStorageIdentity.PRIMARY);
+    static final Event BLOBS_DELETION_EVENT = new FailedBlobEvents.BlobsDeletion(EVENT_ID, BUCKET_NAME, ImmutableList.of(BLOB_ID), ObjectStorageIdentity.PRIMARY);
+    static final Event BUCKET_DELETION_EVENT = new FailedBlobEvents.BucketDeletion(EVENT_ID, BUCKET_NAME, ObjectStorageIdentity.PRIMARY);
+    static final String BlOB_ADDITION_JSON = "{\"type\":\"TmailEventSerializer$BlobAdditionDTO\",\"eventId\":\"6e0dd59d-660e-4d9b-b22f-0354479f47b4\",\"username\":\"secondaryblobstore\",\"bucketName\":\"bucket-1\",\"blobId\":\"blob-id-1\",\"failedObjectStorage\":\"PRIMARY\"}";
+    static final String BlOB_DELETION_JSON = "{\"type\":\"TmailEventSerializer$BlobDeletionDTO\",\"eventId\":\"6e0dd59d-660e-4d9b-b22f-0354479f47b4\",\"username\":\"secondaryblobstore\",\"bucketName\":\"bucket-1\",\"blobId\":\"blob-id-1\",\"failedObjectStorage\":\"PRIMARY\"}";
+    static final String BlOBS_DELETION_JSON = "{\"type\":\"TmailEventSerializer$BlobsDeletionDTO\",\"eventId\":\"6e0dd59d-660e-4d9b-b22f-0354479f47b4\",\"username\":\"secondaryblobstore\",\"bucketName\":\"bucket-1\",\"blobIds\":[\"blob-id-1\"],\"failedObjectStorage\":\"PRIMARY\"}";
+    static final String BUCKET_DELETION_JSON = "{\"type\":\"TmailEventSerializer$BucketDeletionDTO\",\"eventId\":\"6e0dd59d-660e-4d9b-b22f-0354479f47b4\",\"username\":\"secondaryblobstore\",\"bucketName\":\"bucket-1\",\"failedObjectStorage\":\"PRIMARY\"}";
```

Comment:
Please use text blocks here for better clarity, IntelliJ can do the conversion automatically through the "replace with text block" quick-fix.

Reply:
I could not find "replace with text block" in my Intellij. Anyway, it has been updated